assets/chezmoi.io/docs/reference/templates/doppler/doppler.md

```diff
@@ -0,0 +1,15 @@
+# `doppler` *key* [*project* [*config*]]
+
+`doppler` returns the plaintext secret from [Doppler](https://www.doppler.com) using `doppler
+secrets get --plain`.
+
+If any of *project*, or *config* are empty or
+omitted, then chezmoi will use the value from the
+`doppler.project`, and
+`doppler.config` config variables if they are set and not empty.
+
+!!! example
+
+    ```
+    {{ doppler "USERNAME" }}
+    ```
```

Comment:
I think that it could be *worthwhile* mentioning JSON-structured secrets further decoding with `fromJson`. I don’t know that it’s required, but this is one of those complexities of chezmoi in that it is essentially endlessly composable and such compositions may not be always visible.

Reply:
I do mention it in the [user-guide](https://github.com/twpayne/chezmoi/pull/3134/files#diff-fa42ea18298385828ab9d687dc4866fdf6f5018e74ba5b4cd821b6e46edd4a24R50). I didn't think it would be the right place to mention this here - on each function. Otherwise where do you draw that line? Would all new integrations have to list any potential compositions on all function references?  

Thoughts?

Reply:
I had somehow missed that. (small voice)never mind(/small voice)